Update SecretStore API documentation to account for Features feature. (#2863)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-secret-stores/file-secret-store.md

}
```

The flag `multiValued` will decide whether the secret store will present a [name/value behavior or a multiple key-value per secret behavior]({{< ref "secrets_api.md#response-body" >}}).

### Name/Value semantics

> If `multiValued` is `false`, the store will load the file and create a map with the following key value pairs:

Reviewer: ```suggestion
If `multiValued` is `false`, the store will load [the JSON file]({{< ref "#setup-json-file-to-hold-the-secrets" >}}) and create a map with the following key-value pairs:
```

| flattened key           | value                           |
| ---                     | ---                             |
|"redisPassword"          | `"your redis password"`           |
|"connectionStrings:sql"  | `"your sql connection string"`    |
|"connectionStrings:mysql"| `"your mysql connection string"`  |

> With this settings, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message:

Reviewer 2: ```suggestion
With this setting enabled, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message:
```

> With this setting enabled, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message:

Reviewer 2: ```suggestion
With the `multiValued` setting set to true, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message:
```

> With the `multiValued` setting set to true, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message:

Reviewer: ```suggestion
If the `multiValued` setting set to true, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message. For example:
```

```shell
$ curl http://localhost:3501/v1.0/secrets/local-secret-store/connectionStrings
```
```json
{
  "errorCode": "ERR_SECRET_GET",
  "message": "failed getting secret with key connectionStrings from secret store local-secret-store: secret connectionStrings not found"
}
```

> That is the expected behavior as that that key is not present in the table above.

Reviewer 2: ```suggestion
That is the expected behavior as the key is not present in the table above.
```

> That is the expected behavior as the key is not present in the table above.

Reviewer: ```suggestion
This error is expected, since the `connectionStrings` key is not present, per the table above.
```

> On the other hand, requesting for flattened key `connectionStrings:sql` would result in a successful response with the following contents:

Reviewer 2: ```suggestion
On the other hand, requesting for flattened key `connectionStrings:sql` would result in a successful response with the following:
```

> On the other hand, requesting for flattened key `connectionStrings:sql` would result in a successful response with the following:

Reviewer: ```suggestion
However, requesting for flattened key `connectionStrings:sql` would result in a successful response, with the following:
```

```shell
$ curl http://localhost:3501/v1.0/secrets/local-secret-store/connectionStrings:sql
```
```json
{
  "connectionStrings:sql": "your sql connection string"
}
```

### Multiple key-values behavior

> If `multiValued` is `true`, this secret store will present a multiple key-value per secret behavior.

Reviewer 2: ```suggestion
If `multiValued` is `true`, the secret store will present multiple key-value per secret behavior.
```

> If `multiValued` is `true`, the secret store will present multiple key-value per secret behavior.

Reviewer: ```suggestion
If `multiValued` is `true`, the secret store will demonstrate multiple key-value per secret behavior:
```

> Nested structures after the top level will be flattened.

Reviewer: ```suggestion
- Nested structures after the top level will be flattened.
```

> It will parse the same file into this table:

Reviewer: ```suggestion
- It will parse the [same JSON file]({{< ref "#setup-json-file-to-hold-the-secrets" >}}) into this table:
```

| key                | value                           |
| ---                | ---                             |
|"redisPassword"     | `"your redis password"`           |
|"connectionStrings" | `{"mysql":"your mysql connection string","sql":"your sql connection string"}`    |

> Notice here how `connectionStrings` is now a JSON object with two keys: `mysql` and `sql`. Also notice how the two flattened keys `connectionStrings:sql` and `connectionStrings:mysql` are missing from the table above.

Reviewer: ```suggestion
Notice that in the above table:
- `connectionStrings` is now a JSON object with two keys: `mysql` and `sql`. 
- The `connectionStrings:sql` and `connectionStrings:mysql` flattened keys from the [table mapped for name/value semantics]({{< ref "#namevalue-semantics" >}}) are missing.
```

> Invoking a `GET` request on the key `connectionStrings` will result in a successful HTTP response with the following content:

Reviewer: ```suggestion
Invoking a `GET` request on the key `connectionStrings` will now result in a successful HTTP response similar to the following:
```

```shell
$ curl http://localhost:3501/v1.0/secrets/local-secret-store/connectionStrings
```
```json
{
  "sql": "your sql connection string",
  "mysql": "your mysql connection string"
}
```

> Requesting for the flattened key `connectionStrings:sql` would now understandably return in a 500 HTTP error response with the following body:

Reviewer: ```suggestion
Meanwhile, requesting for the flattened key `connectionStrings:sql` would now return a 500 HTTP error response with the following:
```

```json
{
  "errorCode": "ERR_SECRET_GET",
  "message": "failed getting secret with key connectionStrings:sql from secret store local-secret-store: secret connectionStrings:sql not found"
}
```

>  In this example, `connectionStrings` would return the following map:

Reviewer: ```suggestion
```

Should this sentence be removed? Is it referring to the next H4 section?

Author: Yes! Good catch :)

#### Handling deeper nesting levels

> Notice that, as stated in our [Spec metadata fields table](#spec-metadata-fields), `multiValued` only handles a single nesting level.

Reviewer: ```suggestion
Notice that, as stated in the [spec metadata fields table](#spec-metadata-fields), `multiValued` only handles a single nesting level.
```

> Assuming that have a Local File secret store with `multiValued` enabled and pointing to a `secretsFile` with the following JSON content:

Reviewer: ```suggestion
Let's say you have a local file secret store with `multiValued` enabled, pointing to a `secretsFile` with the following JSON content:
```

}
```

> The flag `multiValued` will decide whether the secret store will present a [name/value behavior or a multiple key-value per secret behavior]({{< ref "secrets_api.md#response-body" >}}).

Reviewer 3: ```suggestion
The flag `multiValued` determines whether the secret store presents a [name/value behavior or a multiple key-value per secret behavior]({{< ref "secrets_api.md#response-body" >}}).
```

### Name/Value semantics

> If `multiValued` is `false`, the store will load [the JSON file]({{< ref "#setup-json-file-to-hold-the-secrets" >}}) and create a map with the following key-value pairs:

Reviewer 3: ```suggestion
If `multiValued` is `false`, the store loads [the JSON file]({{< ref "#setup-json-file-to-hold-the-secrets" >}}) and create a map with the following key-value pairs:
```

| flattened key           | value                           |
| ---                     | ---                             |
|"redisPassword"          | `"your redis password"`           |
|"connectionStrings:sql"  | `"your sql connection string"`    |
|"connectionStrings:mysql"| `"your mysql connection string"`  |

> If the `multiValued` setting set to true, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message. For example:

Author: ```suggestion
If the `multiValued` setting set to `false`, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message. For example:
```

> If the `multiValued` setting set to `false`, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message. For example:

Author: ```suggestion
If the `multiValued` setting set to true, invoking a `GET` request on the key `connectionStrings` results in a 500 HTTP response and an error message. For example:
```

> If the `multiValued` setting set to true, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message. For example:

Reviewer 3: Are you sure you mean "true" here? I think you mean "false" for for multiValued here.

Author: you are right. Fixing it..

Author: And commit fixing this sent. Resolving.

> If the `multiValued` setting set to true, invoking a `GET` request on the key `connectionStrings` will result in a 500 HTTP response and an error message. For example:

Reviewer 3: ```suggestion
If the `multiValued` setting set to true, invoking a `GET` request on the key `connectionStrings` results in a 500 HTTP response and an error message. For example:
```

```shell
$ curl http://localhost:3501/v1.0/secrets/local-secret-store/connectionStrings
```
```json
{
  "errorCode": "ERR_SECRET_GET",
  "message": "failed getting secret with key connectionStrings from secret store local-secret-store: secret connectionStrings not found"
}
```

This error is expected, since the `connectionStrings` key is not present, per the table above.

However, requesting for flattened key `connectionStrings:sql` would result in a successful response, with the following:

```shell
$ curl http://localhost:3501/v1.0/secrets/local-secret-store/connectionStrings:sql
```
```json
{
  "connectionStrings:sql": "your sql connection string"
}
```

### Multiple key-values behavior

> If `multiValued` is `true`, the secret store will demonstrate multiple key-value per secret behavior:

Reviewer 3: ```suggestion
If `multiValued` is `true`, the secret store enables multiple key-value per secret behavior:
```

- Nested structures after the top level will be flattened.

> - It will parse the [same JSON file]({{< ref "#setup-json-file-to-hold-the-secrets" >}}) into this table:

Reviewer 3: ```suggestion
- It parses the [same JSON file]({{< ref "#setup-json-file-to-hold-the-secrets" >}}) into this table:
```

| key                | value                           |
| ---                | ---                             |
|"redisPassword"     | `"your redis password"`           |
|"connectionStrings" | `{"mysql":"your mysql connection string","sql":"your sql connection string"}`    |

Notice that in the above table:
- `connectionStrings` is now a JSON object with two keys: `mysql` and `sql`. 
- The `connectionStrings:sql` and `connectionStrings:mysql` flattened keys from the [table mapped for name/value semantics]({{< ref "#namevalue-semantics" >}}) are missing.

> Invoking a `GET` request on the key `connectionStrings` will now result in a successful HTTP response similar to the following:

Reviewer 3: ```suggestion
Invoking a `GET` request on the key `connectionStrings` now results in a successful HTTP response similar to the following:
```

    }
}
```
The contents of key `mysql` under `connectionStrings` has a nesting level greater than 1 and would be flattened.

Here is how it would look in memory:

| key                | value                           |
| ---                | ---                             |
|"redisPassword"     | `"your redis password"`           |
|"connectionStrings" | `{ "mysql:username": "your mysql username", "mysql:password": "your mysql password" }`    |

> Once again, requesting for key `connectionStrings` will result in a successful HTTP response but its contents, as shown in the table above, would be flattened:

Reviewer 3: ```suggestion
Once again, requesting for key `connectionStrings` results in a successful HTTP response but its contents, as shown in the table above, would be flattened:
```
